llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1401,6 +1401,11 @@ InstructionCost RISCVTTIImpl::getCmpSelInstrCost(unsigned Opcode, Type *ValTy,
     return BaseT::getCmpSelInstrCost(Opcode, ValTy, CondTy, VecPred, CostKind,
                                      I);
 
+  // Select might be expanded to move and branch.
```

Comment:
But it also might not...

I think you need to consider availability of the various select variant scalar instructions here and adjust the cost appropriately.  Consider zicond, and whichever of the scalar binmanip added min/max.

Reply:
Thanks for the suggestions. :-) I added `!hasStdExtZicond()` to filter out zicond. For the scalar min/max, it seem the passes called getCmpSelInstrCost will see `llvm.smax.i*`. Assuming they won't change the cost?